File: activity14 Python/gdb/domain/salary_account.py
# gdb/domain/salary_account.py
from gdb.domain.abstract_account import AbstractAccount

class SalaryAccount(AbstractAccount):
    def __init__(self, account_number: str, name: str, age: int, balance: float, status: str = "Active", pin: str = "0000") -> None:
        super().__init__(account_number, name, age, balance, status, pin)

    def calculate_interest(self) -> float:
        return 0.0

    def get_account_type(self) -> str:
        return "Salary"

File: activity14 Java/src/com/gdb/domain/SalaryAccount.java
package com.gdb.domain;

import com.gdb.exceptions.*;

public class SalaryAccount extends AbstractAccount {
    private int inactiveMonths;
    private String employerName;

    public SalaryAccount(String accountNumber, String name, int age, double balance, String status, String pin, String employerName) {
        super(accountNumber, name, age, balance, "SALARY", status, pin);
        this.employerName = employerName;
        this.inactiveMonths = 0;
    }

    @Override
    public void processDebit(double amount) throws AccountException {
        if (amount > this.balance) {
            throw new InsufficientBalanceException("Insufficient funds in Salary account");
        }
        this.balance -= amount;
    }

    public String getEmployerName() { return employerName; }
    public int getInactiveMonths() { return inactiveMonths; }
    public void incrementInactiveMonths() { this.inactiveMonths++; }
}
